components/net/image_cache.rs

```diff
@@ -566,9 +676,111 @@ impl ImageCache for ImageCacheImpl {
         }
     }
 
+    fn add_rasterization_complete_listener(
+        &self,
+        pipeline_id: PipelineId,
+        image_id: PendingImageId,
+        size: DeviceIntSize,
+        sender: IpcSender<ImageCacheMessage>,
+    ) {
+        let mut store = self.store.lock().unwrap();
+        let Some(_) = store.vector_images.get(&image_id).cloned() else {
+            warn!("Unknown image id {image_id:?} requested for rasterization");
+            return;
+        };
+
+        match store.rasterized_vector_images.entry((image_id, size)) {
+            Occupied(mut occupied_entry) => {
+                let task = occupied_entry.get_mut();
+                if task.result.is_some() {
+                    let _ = sender.send(ImageCacheMessage::VectorImageRasterizationCompleted(
+                        pipeline_id,
+                        image_id,
+                        size,
+                    ));
```

Comment:
This looks like it can be done outside of the critical section on `store`, so I would do that.

Reply:
Good point! I've moved the blocking send call outside the critical section in [d6c9182](https://github.com/servo/servo/pull/36721/commits/d6c91826b4885176da8fc5946580d4eea2f0b9ce). Thanks!